app/main/views/direct_award.py

```diff
@@ -0,0 +1,193 @@
+
+from dmapiclient.audit import AuditTypes
+from .. import main
+from ... import db
+from ...models import User, DirectAwardProject, AuditEvent, DirectAwardSearch
+from ...utils import (
+    get_json_from_request, get_int_or_400, json_has_required_keys, pagination_links,
+    get_valid_page_or_1, validate_and_return_updater_request, validate_user_can_access_direct_award_project_or_403)
+
+
+@main.route('/direct-award/projects', methods=['GET'])
+def list_projects():
+    user_id = get_int_or_400(request.args, 'user_id')
+    if user_id is None:
+        abort(400, "User ID not supplied")
+
+    page = get_valid_page_or_1()
+
+    projects = DirectAwardProject.query.filter(DirectAwardProject.users.any(id=user_id))
+    projects = projects.order_by(desc(DirectAwardProject.created_at), desc(DirectAwardProject.id))
+
+    projects = projects.paginate(
+        page=page,
+        per_page=current_app.config['DM_API_PROJECTS_PAGE_SIZE'],
+    )
+
+    return jsonify(
+        projects=[project.serialize() for project in projects.items],
+        meta={
+            "total": projects.total,
+        },
+        links=pagination_links(
+            projects,
+            '.list_projects',
+            request.args
+        ),
+    )
+
+
+@main.route('/direct-award/projects', methods=['POST'])
+def create_project():
+    updater_json = validate_and_return_updater_request()
+
+    json_payload = get_json_from_request()
+    json_has_required_keys(json_payload, ['project'])
+
+    project_json = json_payload['project']
+    json_has_required_keys(project_json, ['name', 'user_id'])
+
+    user = User.query.get(project_json.pop('user_id'))
+
+    if user is None:
+        abort(400, "User ID not supplied")
+
+    project = DirectAwardProject(name=project_json['name'], users=[user])
+    db.session.add(project)
+
+    try:
+        db.session.flush()
+    except IntegrityError as e:
+        db.session.rollback()
+        abort(400, e.orig)
+
+    audit = AuditEvent(
+        audit_type=AuditTypes.create_project,
+        user=updater_json['updated_by'],
+        data={
+            'project_id': project.id,
+            'project_json': project_json,
+        },
+        db_object=project,
+    )
+
+    db.session.add(audit)
+    db.session.commit()
+
+    return jsonify(project=project.serialize()), 201
+
+
+@main.route('/direct-award/projects/<int:project_id>', methods=['GET'])
+def get_project(project_id):
+    user_id = get_int_or_400(request.args, 'user_id')
+    if user_id is None:
+        abort(400, "User ID not supplied")
+
+    validate_user_can_access_direct_award_project_or_403(user_id, project_id)
+
+    return jsonify(project=DirectAwardProject.query.get(project_id).serialize())
+
+
+@main.route('/direct-award/projects/<int:project_id>/searches', methods=['GET'])
+def list_project_searches(project_id):
+    user_id = get_int_or_400(request.args, 'user_id')
+    if user_id is None:
+        abort(400, "User ID not supplied")
```

Comment:
@galund I know what you mean and I definitely have a feeling that the checks in the frontend can be unreliable since they're easy to forget without noticing anything's wrong.

The main problem with GET is that it goes against the read behaviours we expect from the API:
1. It's not obvious what's the best way to list project searches with a script for example since we'd need to attach a user_id for each request, which isn't actually what's happening in that case.
2. In pretty much every other list endpoint I'd read the `user_id` GET argument as "only return objects from this user". This goes against that and makes it hard to combine with that filter (or any other filter) in the same endpoint.
3. GET parameter just feels like a wrong place for authentication details in terms of caching, logging access etc.

I think if we extend this to "how would this look in a public API" it becomes clear that API would have to do both authentication and authorization for this and other read and write endpoints that are currently doing this in the frontend apps (eg draft briefs, draft services). At that point, we'd need to find a way to securely transfer authentication details to the API and authorize everywhere based on that. Since we haven't thought about what that would look like and whether we'd want to use it in our own frontends as well I think we should stick with the current common approach for now and do the checks in the frontends.